src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1851,6 +1851,26 @@ static void addFilterScope(const Scope* scoP, std::vector<BSONObj> &filters)
   }
 }
 
+/* ****************************************************************************
+*
+* sortCriteria -
+*
+*/
+static std::string sortCriteria(const std::string& sortToken)
+{
+  if (sortToken == DATE_CREATED)
+  {
+    return std::string(ENT_CREATION_DATE);
+  }
+
+  if (sortToken == DATE_MODIFIED)
+  {
+    return std::string(ENT_MODIFICATION_DATE);
```

Comment:
std::string should not be needed here.
However, if you prefer to have it, no problem.

I would make these first ifs one-liners:

    if      (sortToken == DATE_CREATED)   return ENT_CREATION_DATE;
    else if (sortToken == DATE_MODIFIED)  return ENT_MODIFICATION_DATE;
   
NTC